packages/server/graphql/types/Discussion.ts

```diff
@@ -42,6 +49,54 @@ const Discussion = new GraphQLObjectType<any, GQLContext>({
       description:
         'The partial foregin key that describes the type of object that is the topic of the discussion. E.g. AgendaItem, TaskId, ReflectionGroup, GitHubIssue'
     },
+    stage: {
+      type: NewMeetingStage,
+      description: 'The stage that the discussion is located',
+      resolve: async (
+        {discussionTopicId, discussionTopicType, meetingId},
+        _args: unknown,
+        {dataLoader}
+      ) => {
+        const meeting = await dataLoader.get('newMeetings').load(meetingId)
+        const {phases, teamId} = meeting
+        switch (discussionTopicType) {
+          case 'agendaItem': {
```

Comment:
I took a stab at reducing duplication between these cases, but gave up after failing to wrangle the types and templating :/